### ISSUE TITLE
Plugin doesn't work with Teamcity 10.0.2

### ISSUE THREAD

**Issue by davestephens** [parautenbach/TeamCity-HipChat-Notifier#78]
Since upgrading to TC10, the notifier fails to connect to Hipchat hosted server (I've not tested Hipchat "proper").

The API is definitely enabled:

https://chat/v2/health-check:

```
{
status: "okay"
}
```

Yet when hitting **Test Connection** we get a "Connection Failed" error message.

I've also regenerated the API key just in case, and it's not that either...

**Comment by dvdgutzmann** [comment]
Same here. "Connection Failed" on Test Connection. TC10.

**Comment by barretb** [comment]
I'm also getting this message with Team City 9.1.6, plug in version 0.9.2. It started around mid-September. I have tried regenerating my token a couple of times (tried both admin tokens and notification tokens), but I still get this error, and nothing else. And I can't find any trace of an error message in the Team City server logs either, which isn't helping me track down the cause.

Test connection fails, can't get rooms list, anything. I have verified that I can connect to api.hipchat.com from my Team City machine as well, so it's not a proxy, firewall or internet issue, and it was working fine up until that point. I made no update to either Team City or the plugin, so that shouldn't be the culprit of it ceasing to work.

**Comment by davidkeaveny** [comment]
I'm getting the same with TeamCity 10.0.0, connecting to HipChat Cloud with the 0.9.2 plugin. When configuring, the Test Connection button reports "failed", the Rooms dropdowns are not populating.

**Comment by SteveLobdell** [comment]
I just uploaded the zip to 10.0.1 Enterprise and restarted TC.  The zip changed to a file with no extension which I'm guessing means that TC installed or tried to.  But now I'm not seeing it in the plugins list and there's no Hipchat options on the admin side or in any projects.  Am I missing something?

**Comment by davestephens** [comment]
I had 30 mins to play about with this at work earlier this week, I got the
same on a fresh TeamCity install.

Where the plugin had been working on a TC9 install and subsequently
upgraded to TC10, you get all the UI bits and bobs but it doesn't work
properly.

TL;DR; it's not compatible with TC10 in any way whatsoever. :-)

**Comment by Funbit** [comment]
Is there any update on this? Such a great plugin but without TC 10 support :(

**Comment by sderderian** [comment]
Is there any update on this issue?  Have end users inquiring.....

**Comment by davestephens** [comment]
@sderderian You could always raise a pull request...

**Comment by davestephens** [comment]
For those interested I've fixed the plugin and released it from my fork: https://github.com/ManDevOps/TeamCity-HipChat-Notifier/releases/tag/0.9.3

I've tested it against TC10 and 2017.1, and it seems to work OK - as such I'm closing this issue.

**Comment by parautenbach** [comment]
Hello @davestephens 

Thanks for this. I just don't currently have enough spare time to code on this. I try every now and then to make some time, as there are a few features I'd still like to implement one day. 

I'd be more than happy to accept patches and PRs though! No need to fork this (but it's fine if you want to): You're welcome to submit a PR and the latest ZIP, with an updated README and tests, and I'll publish it.

**Comment by davestephens** [comment]
Hi @parautenbach, 

I've knocked this out at work so also don't have loads of time to spend on it - I'm using an internal HipChat server so can't verify the external tests all pass against HipChat 'proper' as it's blocked in my office. If I get some time I'll tidy it up at home into a proper PR and submit it!
